lib/models/patch/multi-file-patch.js

```diff
@@ -332,6 +336,13 @@ export default class MultiFilePatch {
     for (const hunk of filePatch.getHunks()) {
       this.hunksByMarker.set(hunk.getMarker(), hunk);
     }
+
+    const diffRowOffsetIndex = this.diffRowOffsetIndices.get(filePatch.getPath());
+    // if the patch was initially collapsed, we need to calculate
+    // the diffRowOffsetIndices to calculate comment position.
+    if (diffRowOffsetIndex.index.size === 0) {
```

Comment:
🎨minor readability suggestion -- I noticed I had to look at this a couple times, as well as `assert.strictEqual(multiFilePatch.diffRowOffsetIndices.get('1.txt').index.size, 0)` in the tests.

Perhaps we could add a method `diffRowOffsetIndexUnpopulated` that does this check and reads more declaratively. And we can also use the method to improve readability in our test assertions.

Reply:
Hmm is there an edge case here for symlink and file mode change diffs that would have empty indexes? I guess we'd just unnecessarily call `populateDiffRowOffsetIndices` and create a new empty index, which isn't a big deal. It's probably not worth the extra effort to distinguish between empty indexes from large diffs versus symlink or file mode changes, huh?

Reply:
yeah, I'd take readability and simplicity over dealing with symlink and file mode changes here.  Creating a new empty index shouldn't impact performance very much.

I'll work on the other changes though, thanks for the suggestion.